tests/gateways/test_connection.py

```diff
@@ -106,7 +107,60 @@ def test_s3_server(minio_s3_server):
             boto3.session.Session.resource, "__defaults__", patched_defaults
         ):
             # the .conda files in this repo are somehow corrupted
-            with env_var("CONDA_USE_ONLY_TAR_BZ2", "True"):
+            with env_vars(
+                {"CONDA_USE_ONLY_TAR_BZ2": "True", "CONDA_SUBDIR": "linux-64"}
+            ):
+                with make_temp_env(
+                    "--override-channels",
+                    f"--channel=s3://{bucket_name}",
+                    "--download-only",
+                    "--no-deps",  # this fake repo only includes the zlib tarball
+                    "zlib",
+                    use_exception_handler=False,
+                    no_capture=True,
+                ):
+                    # we just want to run make_temp_env and cleanup after
+                    pass
+
+
+@pytest.mark.integration
+def test_s3_server_with_mock(package_server_ssl):
+    """
+    Use boto3 to fetch from a mock s3 server pointing at the test package
+    repository. This works since conda only GET's against s3 and s3 is http.
+    """
+    import boto3
+    from botocore.client import Config
+
+    host, port = package_server_ssl.getsockname()
+    endpoint_url = f"https://{host}:{port}"
+    bucket_name = "test"
+
+    # We patch the default kwargs values in boto3.session.Session.resource(...)
+    # which is used in conda.gateways.connection.s3.S3Adapter to initialize the S3
+    # connection; otherwise it would default to a real AWS instance
+    patched_defaults = (
+        "us-east-1",  # region_name
+        None,  # api_version
+        True,  # use_ssl
+        False,  # verify
+        endpoint_url,  # endpoint_url
+        "minioadmin",  # aws_access_key_id
+        "minioadmin",  # aws_secret_access_key
+        None,  # aws_session_token
+        Config(signature_version="s3v4"),  # config
+    )
+    with pytest.raises(CondaExitZero):
+        with patch.object(
+            boto3.session.Session.resource, "__defaults__", patched_defaults
```

Comment:
This patch is the same as in `test_s3_server` using minio except for the url. It is used to point boto3 at a non-AWS server.